src/SFA.DAS.Data.AcceptanceTests/DataLockEventTests/WhenADataLockEventIsCreated.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using SFA.DAS.Data.Tests.Builders;
+using SFA.DAS.Provider.Events.Api.Types;
+
+namespace SFA.DAS.Data.AcceptanceTests.DataLockEventTests
+{
+    [TestFixture]
+    public class WhenADataLockEventIsCreated : DataLockEventTestsBase
+    {
+        [Test]
+        public async Task ThenTheDataLockEventsAreStored()
+        {
+            var expectedDataLockEvents = ConfigureEventsApi();
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            var cancellationToken = cancellationTokenSource.Token;
+            Task.Run(() => WorkerRole.Run(), cancellationToken);
+
+            var databaseAsExpected = TestHelper.ConditionMet(IsDatabaseInExpectedState, TimeSpan.FromSeconds(60));
+
+            cancellationTokenSource.Cancel();
+            Assert.IsTrue(databaseAsExpected);
+
+            await AreDataLockFieldsStoredCorrectly(expectedDataLockEvents);
+        }
+
+        private async Task AreDataLockFieldsStoredCorrectly(IList<DataLockEvent> expectedDataLockEvents)
+        {
+            var actualDataLocks = (await EventTestsRepository.GetDataLocks()).OrderBy(d => d.DataLockId).ToList();
+            var actualDataLockErrors = (await EventTestsRepository.GetDataLockErrors()).OrderBy(e => e.DataLockId).ThenBy(e => e.ErrorCode).ToList();
+
+            for (var i = 0; i < actualDataLocks.Count; i++)
+            {
+                Assert.AreEqual(expectedDataLockEvents[i].Id, actualDataLocks[i].DataLockId);
+
+                var expectedDataLock = expectedDataLockEvents[i];
+                var actualDataLock = actualDataLocks[i];
+
+                Assert.AreEqual(expectedDataLock.ProcessDateTime, actualDataLock.ProcessDateTime);
```

Comment:
FluentAssertions can compare 2 objects to make sure the property values are the same (I think!), would be easier than checking each individual property?

Reply:
```
actualDataLock
.Should()
.BeEuivalentTo(expectedDataLock)

```

Is the one I think. Uses default equality comparison for property types. 


Also if we are tidying up in that area, it looks like we could do some collection assertions as well. 

Will actually have a look at this (when VS and Slack stop killing my machine)